common/src/main/java/org/opensearch/ml/common/transport/connector/MLConnectorDeleteRequest.java

```diff
@@ -23,22 +24,34 @@
 
 public class MLConnectorDeleteRequest extends ActionRequest {
     @Getter
-    String connectorId;
+    private final String connectorId;
+    @Getter
+    private final String tenantId;
 
-    @Builder
     public MLConnectorDeleteRequest(String connectorId) {
+
+        this.connectorId = connectorId;
+        this.tenantId = null;
+    }
+
+    @Builder
+    public MLConnectorDeleteRequest(String connectorId, String tenantId) {
+
         this.connectorId = connectorId;
+        this.tenantId = tenantId;
     }
 
     public MLConnectorDeleteRequest(StreamInput input) throws IOException {
         super(input);
         this.connectorId = input.readString();
+        this.tenantId = input.readOptionalString();
```

Comment:
If this code will also used for OS, should add BWC version check?

Reply:
Nice catch, @dhrubo-os can you test and double confirm?

Reply:
I created a separate task for BWC functionality for tenantId. I'm thinking based on when are we going to release this feature, I'll make a separate PR dedicated for BWC cases.